pkg/decoder/decoder.go

```diff
@@ -7,65 +7,84 @@ package decoder
 
 import (
 	"bytes"
-	"errors"
-	"regexp"
-	"time"
 
 	"github.com/DataDog/datadog-log-agent/pkg/config"
-	"github.com/DataDog/datadog-log-agent/pkg/message"
 )
 
-// countDownDuration represents the time after which the buffered line is sent when there is no more incoming data
-const countDownDuration = 1 * time.Second
-
 // maxMessageLen represents the maximum length for a message
-var maxMessageLen = config.MaxMessageLen
+var maxMessageLen = config.MaxMessageLen - len([]byte(config.TruncWarningMsg))
+
+// Payload represents a list of bytes consumed by the Decoder
+type Payload struct {
+	content []byte
+}
+
+// NewPayload returns a new decoder payload
+func NewPayload(content []byte) *Payload {
+	return &Payload{content}
+}
+
+// Message represents a list of bytes produced by the Decoder
+type Message struct {
```

Comment:
This is getting confusing, could we rename into `DecoderInput` and `DecoderOutput` maybe?